tests/lightning_tensor/test_serialize_tensor.py

```diff
@@ -0,0 +1,936 @@
+        mpsHermitianObsC128,
+        mpsNamedObsC64,
+        mpsNamedObsC128,
+        mpsTensorProdObsC64,
+        mpsTensorProdObsC128,
+    )
+
+
+@pytest.fixture(params=["mps", "exact"])
+def tn_backend(request):
+    return request.param
+
+
+def get_module_name(tn_back, name, dtype):
+    mod_name = tn_back + name
+    mod_name = mod_name + "64" if dtype else mod_name + "128"
+    return globals().get(mod_name)
+
+
+def test_wrong_device_name():
+    """Test the device name is not a valid option"""
+
+    with pytest.raises(qml.DeviceError, match="The device name"):
+        QuantumScriptSerializer("thunder.qubit")
+
+
+@pytest.mark.parametrize("dtype", ["64", "128"])
+@pytest.mark.parametrize(
+    "obs,obs_type",
+    [
+        (qml.PauliZ(0), "NamedObsC"),
+        (qml.PauliZ(0) @ qml.PauliZ(1), "TensorProdObsC"),
+        (qml.Hadamard(0), "NamedObsC"),
+        (qml.Hermitian(np.eye(2), wires=0), "HermitianObsC"),
+        ((qml.PauliZ(0) @ qml.Hadamard(1) @ qml.PauliZ(2) @ qml.PauliX(3)), "TensorProdObsC"),
+        (
+            qml.PauliZ(0) @ qml.PauliY(1) @ qml.PauliX(2),
+            "TensorProdObsC",
+        ),
+        (
+            qml.PauliZ(0) @ qml.PauliY(1) @ (0.1 * (qml.PauliZ(2) + qml.PauliX(3))),
+            "HamiltonianC",
+        ),
+        (
+            (
+                qml.Hermitian(np.eye(2), wires=0)
+                @ qml.Hermitian(np.eye(2), wires=1)
+                @ qml.Projector([0], wires=2)
+            ),
+            "TensorProdObsC",
+        ),
+        (
+            (qml.Hermitian(np.eye(2), wires=0)),
+            "HermitianObsC",
+        ),
+        (
+            qml.PauliZ(0) @ qml.Hermitian(np.eye(2), wires=1) @ qml.Projector([0], wires=2),
+            "TensorProdObsC",
+        ),
+        (qml.Projector([0], wires=0), "HermitianObsC"),
+        (qml.Hamiltonian([1], [qml.PauliZ(0)]), "NamedObsC"),
+        (qml.sum(qml.Hadamard(0), qml.PauliX(1)), "HamiltonianC"),
+        (
+            (0.5 * qml.PauliX(0)),
+            "HamiltonianC",
+        ),
+        (2.5 * qml.PauliZ(0), "HamiltonianC"),
+    ],
+)
+def test_obs_returns_expected_type(tn_backend, dtype, obs, obs_type):
+    """Tests that observables get serialized to the expected type, with and without wires map"""
+
+    mod_name = tn_backend + obs_type + dtype
+    obs_type_mod = globals().get(mod_name)
+
+    tn_method = tn_backend
+    tn_method = "tn" if tn_method == "exact" else tn_method
+
+    serializer = QuantumScriptSerializer(
+        device_name, True if dtype == "64" else False, tensor_backend=tn_method
+    )
+    assert isinstance(serializer._ob(obs, dict(enumerate(obs.wires))), obs_type_mod)
+    assert isinstance(serializer._ob(obs), obs_type_mod)
+
+
+wires_dict = {i: i for i in range(10)}
+
+
+@pytest.mark.parametrize("use_csingle", [True, False])
+@pytest.mark.parametrize("wires_map", [wires_dict, None])
+class TestSerializeObs:
+    """Tests for the _observables function"""
+
```

Comment:
You can probably make tn_method a class attribute, and call it `self.tn_method` everywhere. Without inputting the fixture tn_backend in each function.

Reply:
You can even make `tn_backend` a class attribute too, if needed.

Reply:
Thank you. I changed it to class attribute